Blog post for IAM launch (#15226)
pulumi/docs: content/blog/pulumi-cloud-iam-launch/index.md

## Launching Today: Granular Access Tokens via Custom Roles

This vision begins today with the initial phase of Pulumi IAM, enabling you to define **Custom Roles** built from **fine-grained Permissions** and apply them specifically to **Organization Access Tokens**. This initial step provides immediate, significant security benefits, particularly for automation:

*   **True Least Privilege for CI/CD:** Scope pipeline tokens to *only* the actions (e.g., `pulumi up`) and Entities (e.g., `stack: myapp-prod`) they absolutely need.
*   **Reduced Blast Radius:** If a scoped token is compromised, the potential damage is limited strictly to the permissions defined in its associated role.
*   **Enhanced Compliance:** Demonstrate precise control over programmatic access to auditors.

## How to Get Started with Granular Access Tokens

Configuring and using Custom Roles for scoped tokens is done via the Pulumi Cloud console:

#### 1. Define a Custom Permission Set (Optional)

* Create reusable sets of fine-grained permissions. Navigate to Organization Settings -> Roles & Permissions -> Permissions.*

> * Click "Create", name it (e.g., "Stack Update Only"), select Resource Type (e.g., Stack), check specific permissions (`stack:read`, `stack:update`), and save.

Reviewer: super nitpick but the [style guide](https://github.com/pulumi/docs/blob/master/STYLE-GUIDE.md) suggests using `Select` in place of `Click`
```suggestion
* Select "Create", name it (e.g., "Stack Update Only"), select Resource Type (e.g., Stack), check specific permissions (`stack:read`, `stack:update`), and save.
```

#### 2. Create a Custom Role

* Combine permissions with specific resources. Navigate to Organization Settings -> Roles & Permissions -> Roles.*

> * Click "Create", name it (e.g., "CICD-ProdApp-Deployer"). Add policies: select Resource Type (e.g., Stacks), choose a Permission Set (standard or custom), specify target Stacks, and save. Add more policies for different resource types (Environments, etc.) as needed.

Reviewer: Same thing here re. select vs click

```suggestion
* Select "Create", name it (e.g., "CICD-ProdApp-Deployer"). Add policies: select Resource Type (e.g., Stacks), choose a Permission Set (standard or custom), specify target Stacks, and save. Add more policies for different resource types (Environments, etc.) as needed.
```

{{% notes type="info" %}}
**Available Today:** Custom Permissions, Custom Roles, and the ability to scope Organization Access Tokens using these roles, is **available now** for customers on the **Pulumi Enterprise** and **Pulumi Business Critical** tiers. Explore these features in your Pulumi Cloud organization settings!
{{% /notes %}}

## Conclusion: Building a More Secure Future

Pulumi Identity and Access Management (IAM) represents a fundamental advancement in securing cloud development lifecycles managed by Pulumi, providing the controls needed to confidently embrace speed and scale. Today’s launch of Granular Access Tokens via Custom Roles provides immediate security improvements for automation and programmatic access, laying the vital groundwork for our comprehensive IAM vision rooted in Zero Trust principles.

This empowers platform and security teams with the fine-grained control needed to implement least privilege, enhance compliance, and scale Pulumi usage securely without sacrificing velocity.

> We encourage our Enterprise and Business Critical customers to explore Custom Roles and Granular Access Tokens today. Dive into the [documentation]() (link placeholder) and start building roles tailored to your security requirements. We welcome your feedback and feature requests in our [GitHub repository](https://github.com/pulumi/pulumi-service/issues) (link placeholder). Join us as we build a more secure foundation for cloud engineering!

Reviewer: The link placeholders need to get updated/deleted:

```suggestion
We encourage our Enterprise and Business Critical customers to explore Custom Roles and Granular Access Tokens today. Dive into the [documentation](/docs/pulumi-cloud/access-management/rbac) and start building roles tailored to your security requirements. We welcome your feedback and feature requests in our [GitHub repository](https://github.com/pulumi/pulumi-service/issues). Join us as we build a more secure foundation for cloud engineering!
```
